site-src/geps/gep-709.md

```diff
@@ -0,0 +1,260 @@
+resource that enables app admins to describe where they trust references from.
+
+## Motivation/User Journeys/Background
+
+This GEP enables the following capabilities:
+
+1. Retaining full control of Gateway and Routes in an infra namespace, while
+   targeting apps in different namespaces.
+1. Traffic splitting between Services in different namespaces.
+1. Mesh overrides to target Services in different namespaces. 
+
+## ReferencePolicy
+
+Anytime we allow crossing a namespace boundary, we need to be very cautious. To
+ensure that this feature is safe, we need to enforce a handshake mechanism that
+requires resources in both namespaces to agree to this reference. To accomplish
+that, a new ReferencePolicy resource should be introduced.
+
+![Reference Policy](images/709-referencepolicy.png)
+
+With this model, Routes would be able to directly reference Routes and Services
+in other namespaces. These references would only be considered valid if a
+ReferencePolicy in the target namespace explicitly allowed it. 
+
+The following example shows how a HTTPRoute in namespace foo could reference
+a Service in namespace bar. In this example a ReferencePolicy in the bar
+namespace explicitly allows references to Services from HTTPRoutes in the foo
+namespace.
+
+```yaml
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+spec:
+  rules:
+  - matches:
+    - path: /bar
+    forwardTo:
+      backend:
+      - name: bar
+        namespace: bar
+---
+kind: ReferencePolicy
+metadata:
+  name: bar
+  namespace: bar
+spec:
+  from:
+  - group: networking.gateway.k8s.io
+    kind: HTTPRoute
+    namespace: foo
+  to:
+  - group: core
+    kind: Service
+```
+
+### API
+This proposed API is fairly straightforward, but comes with a couple notable
+decisions:
+
+1. Each ReferencePolicy only supports a single From and To section. Additional
+   trust relationships can be modeled with additional ReferencePolicy resources.
+1. Resource names are intentionally excluded from this policy for simplicity and
+   because they rarely provide any meaningful protection.
+1. A single Namespace is allowed per "From" struct. Although a selector would be
+   more powerful it may encourage unnecessarily insecure configuration.
+
+```go
+// ReferencePolicy identifies cross namespace relationships that are trusted for
+// Gateway API.
+type ReferencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of ReferencePolicy.
+    Spec ReferencePolicySpec `json:"spec,omitempty"`
+}
+
+
+// ReferencePolicySpec identifies a cross namespace relationship that is trusted
+// for Gateway API.
+type ReferencePolicySpec struct {
+    // From describes the trusted namespaces and kinds that can reference the
+    // resources described in "To".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    From []ReferencePolicyFrom `json:"from"`
```

Comment:
Both `from` and `to` are slices, but we don't have anything describing how multiple entries in the slice must be combined.

Seems like it would probably be something like "Each entry in `from` must be considered to be an additional place that references can be valid from, or to put this another way, entries must be combined using OR."

Similarly for the `to` stanza, I think that combining entries with OR is the only thing that makes sense. But regardless, we should have the behavior documented here.

Reply:
Good catch, this was not well defined. Added your suggestion verbatim.